app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt

```diff
@@ -60,20 +67,28 @@ class MessagingService : FirebaseMessagingService() {
                     )
                 }
             }
-        }
 
-        remoteMessage.notification?.let {
-            if (it.body == "request_location_update") {
+            if (!it.containsKey(MESSAGE)) {
+                Log.e(TAG, "Message missing from notification.")
+                return
+            }
+
+            val title = it[TITLE]
+            val message = it[MESSAGE]!!
```

Comment:
Why break the app if it's `null`?

```suggestion
            val message = it[MESSAGE] ?: ""
```

Reply:
The message is a required field from HA.  If it doesn't make it to the application we have a problem.

https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/notify/__init__.py#L45